### ISSUE TITLE
Make speedreader pages article complaint

### ISSUE THREAD

**Issue by nullhook** [brave/brave-browser#30346]
The current speedreader extracts all content nodes inside the <body> and adds an id of "article". However, since we're not using the real `<article>` element, the page cannot be inferred as an article by accessibility or distillers. Major news websites typically follow a specific structure.

To address this issue with distilling content from accessibility nodes using the AXTreeDistiller interface in AIChat, we should change the markup of the speedreader pages to conform to the following structure.

```
<main>
<article>...</article>
</main>
```
